docs/source/examples/dataframe-hdf5.rst

```diff
@@ -0,0 +1,146 @@
+
+.. code-block:: python
+   :emphasize-lines: 40,45
+   
+    
+    def create_hdf5s(iters):
+           
+       import random
+       import pandas as pd
+       import numpy as np
+       import string 
+       import json
+    
+       # Intitalize counter to build specific number of hdf5 files
+       count = 0
+       setseed=1
+       np.random.seed(seed=setseed)
+       random.seed(setseed)
+    
+       # Key to store group keys and file names for generated hdf5 files
+       fileKeys = {}
+    
+       while count <= iters:
+            
+           # randomly pick letter as dataset key
+           groupkey = random.choice(list(string.ascii_lowercase))
+        
+           # randomly pick a number as hdf5 filename
+           saver = 'my'+str(np.random.randint(100))+'.h5'
+        
+           # Make a dataframe; 26 rows, 2 columns
+           df = pd.DataFrame(
+           {'x': np.random.randint(1,1000,26),
+           'y': np.random.randint(1,1000,26)},
+           index=list(string.ascii_lowercase)
+           )
+        
+           # Write synthetic hdf5 to current directory
+           df.to_hdf(saver, key='/'+groupkey, format='table')
+           fileKeys[saver]=groupkey
+           count += 1
+           setseed+=1
+        
+       # Return the dictionary with file key and filename for testing
+       with open('hdf5Keys.json','w') as f:
+           f.write(json.dumps(fileKeys))
+
+    return fileKeys
+
+    # Create the HDF5 files and keys to access
+    hdf5Keys = create_hdf5s(keys)
+
+
+.. doctest:: 
+    
+    def create_hdf5s(iters):
+        import random
+        import pandas as pd
+        import numpy as np
+        import string 
+        import json
+
+        # Intitalize counter to build specific number of hdf5 files
+        count = 0
+
+        # Key to store group keys and file names for generated hdf5 files
+        fileKeys = {}
+
+        while count <= iters:
+            # randomly pick letter as dataset key
+            groupkey = random.choice(list(string.ascii_lowercase))
+
+            # randomly pick a number as hdf5 filename
+            saver = 'my'+str(np.random.randint(100))+'.h5'
+
+            # Make a dataframe; 26 rows, 2 columns
+            df = pd.DataFrame(
+            {'x': np.random.randint(1,1000,26),
+            'y': np.random.randint(1,1000,26)},
+            index=list(string.ascii_lowercase)
+            )
+
+            # Write synthetic hdf5 to current directory
+            df.to_hdf(saver, key='/'+groupkey, format='table')
+
+            fileKeys[saver]=groupkey
+
+            count += 1
+        # Return the dictionary with file key and filename for testing
+        with open('hdf5Keys.json','w') as f:
+            f.write(json.dumps(hdf5Keys))
+        return fileKeys
+    
+    # Build the exercise data
+    hdf5Keys = create_hdf5s(keys)
+        
+**Read single dataset from HDF5:**
```

Comment:
@shoyer  
> users who didn't write their HDF5 files with pandas will almost certainly be unable to read them with dask.dataframe.

NOW IT MAKES SENSE!  I tried writing a synthetic HDF5 file with h5py and never got `dask.dataframe` to read it in.  Then, I switched to what I knew (pandas) and got my hdf5 file.  It could be operator error (h5py), but after researching your suggestion, I see why it's important to include this note on PyTables flavored HDF5.  .  

Another challenge is, I see some examples using h5py to create "synthetic hdf5" files that are used for examples.  Should we also remove any references to `h5py` built files while also adding the emphasis you suggest?